pkg/resource/apply.go

```diff
@@ -41,21 +47,24 @@ const fieldTooLong metav1.CauseType = "FieldValueTooLong"
 
 // Apply applies the given resource bytes to the target cluster specified by kubeconfig
 func (r *helper) Apply(obj []byte) (ApplyResult, error) {
-	factory, err := r.getFactory("")
-	if err != nil {
-		r.logger.WithError(err).Error("failed to obtain factory for apply")
-		return "", err
-	}
 	ioStreams := genericclioptions.IOStreams{
 		In:     &bytes.Buffer{},
 		Out:    &bytes.Buffer{},
 		ErrOut: &bytes.Buffer{},
 	}
-	applyOptions, changeTracker, err := r.setupApplyCommand(factory, obj, ioStreams)
+	applyOptions, changeTracker, err := r.setupApplyCommand(obj, ioStreams)
 	if err != nil {
 		r.logger.WithError(err).Error("failed to setup apply command")
 		return "", err
 	}
+
+	// fakeApply mode (used in scale testing only) doesn't run the apply. Instead it will Sleep for a period of time
+	// that matches our request time metrics in real hive environments to simulate i/o wait.
+	fakeAppliesEnvVar, _ := os.LookupEnv(constants.HiveSyncSetsFakeApplyEnvVar)
+	if fakeApplies, err := strconv.ParseBool(fakeAppliesEnvVar); fakeApplies && err == nil {
```

Comment:
should the env var and and ParseBool be called when constructing the resource helper instead of every apply?